media/js/wiki.js

```diff
@@ -847,5 +847,73 @@
     if($('details').length){
         initDetailsTags();
     }
+    
+    function create_jsfiddle(title, htmlCode, cssCode, jsCode) {
+       var $form = $('<form method="post" target="_blank" action="https://jsfiddle.net/api/post/library/pure/">'
+            + '<input type="hidden" id="html" name="html" />'
+            + '<input type="hidden" id="css" name="css" />'
+            + '<input type="hidden" id="js" name="js" />'
+            + '<input type="hidden" id="panel_html" name="panel_html" value="0" />'
+            + '<input type="hidden" id="panel_css" name="panel_css" value="0" />'
+            + '<input type="hidden" id="panel_js  name="panel_js value="0" />'
+            + '<input type="hidden" id="title" name="title" value="' + title +'" />'
+            + '<input type="hidden" id="normalize_css" name="normalize_css" value="0" />'
+            + '<input type="hidden" id="wrap" name="wrap" value="d" />'
+            + '<input type="submit" />'
+        + '</form>');
+       $form.find('#html').val(htmlCode);
+       $form.find('#css').val(cssCode);
+       $form.find('#js').val(jsCode);
+       $form.submit();
+    }
+    
+    function create_codepen(title, htmlCode, cssCode, jsCode) {
+       var $form = $('<form method="post" target="_blank" action="http://codepen.io/pen/define">'
+            + '<input type="hidden" id="data" name="data">'
+            + '<input type="submit" />'
+        + '</form>');
+       var data = {'title': title, 'html': htmlCode, 'css': cssCode, 'js': jsCode};
+       $form.find('#data').val(JSON.stringify(data));
+       $form.submit();
+    }
+           
+    function create_sample(sampleCodeHost, title, htmlCode, cssCode, jsCode) {
+       if(sampleCodeHost == 'jsfiddle') {
+            create_jsfiddle(title, htmlCode, cssCode, jsCode);
+        } else if(sampleCodeHost == 'codepen') {
+            create_codepen(title, htmlCode, cssCode, jsCode);
+        } 
+    }
+
+    $('#wikiArticle').on('click', 'a.open-in-host', function(){
```

Comment:
Since these links will not work without javascript enabled we should use the HTML button element instead of links. (links without the href element cannot receive keyboard focus).